libbeat/management/management.go

```diff
@@ -74,22 +74,37 @@ type Manager interface {
 	// Enabled returns true if manager is enabled.
 	Enabled() bool
 
-	// Start the config manager giving it a stopFunc callback
-	// so the beat can be told when to stop.
-	Start(stopFunc func())
-
-	// Stop the config manager.
+	// Start needs to be invoked when the system is ready to receive an external configuration and
+	// also ready to start ingesting new events. The manager expects that all the reloadable and
+	// reloadable list are fixed for the whole lifetime of the manager.
+	//
+	// Notes: Adding dynamically new reloadable hooks at runtime can lead to inconsistency in the
+	// execution.
+	Start() error
+
+	// Stop when this method is called, the manager will stop receiving new actions, no more action
+	// will be progated to the handlers and will not try to configure any reloadable parts.
+	// When the manager is stopped the callback will be called to signal that the system can terminate.
+	//
+	// Calls to 'CheckRawConfig()' or 'SetPayload()' will be ignored after calling stop.
+	//
+	// Note: Stop will not call 'UnregisterAction()' automaticallty.
 	Stop()
 
+	// SetStopCallback accepts a function that need to be called when the manager want to shutdown the
+	// beats.
+	SetStopCallback(f func())
```

Comment:
When should a stop function be registered via this callback rather than providing the functionality in the [Beater `Stop()`](https://github.com/elastic/beats/blob/main/libbeat/beat/beat.go#L48-L50) function?